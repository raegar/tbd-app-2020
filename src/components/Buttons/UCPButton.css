.finalbutton{
  background-color: #005C6E;
  color: black;
  width: 100px;
  height: 50px;
  font-size: large;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  border-radius: 25px;
  border: 2px solid black;
}

.smallbutton{
  background-color: #31B7BC;
  color: black;
  width: 100px;
  height: 50px;
  font-family: 'Montserrat', sans-serif;
  font-size: 16px;    border-radius: 25px;
  border: 2px solid #005C6E;
  outline: none !important;
}

.mediumbutton{
  background-color: #31B7BC;
  color: black;
  width: 125px;
  height: 75px;
  font-family: 'Montserrat', sans-serif;
  font-size: 16px;    border-radius: 25px;
  border: 2px solid #005C6E;
  outline: none !important;
}

.largebutton{
    background-color: #31B7BC;
    color: black;
    width: 200px;
    height: 100px; 
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;
}

.extralargebutton{
    background-color: #31B7BC;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;
}

.admin-navbar-button{
  background-color: #005C6E;
  color: #797979;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  border-radius: 0;
  border: 0;
  outline: none !important;
}

.admin-navbar-button:hover{
  background-color: #005C6E;
  color: #D5D5D5;
}

.admin-navbar-button-active{
  background-color: #fff;
  color: #797979;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  border-radius: 0;
  border: 0;
  outline: none !important;
}

  .smallbutton:hover{
    background-color: white;
    color: black;
    width: 100px;
    height: 50px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }

  .mediumbuttom:hover{
    background-color: white;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }

  .largebutton:hover{
    background-color: white;
    color: black;
    width: 200px;
    height: 100px; 
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }
  .extralargebutton:hover{
    background-color: white;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }
