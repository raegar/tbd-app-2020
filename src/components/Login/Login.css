.UserInfo {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 90%;
}

.UserInfo span {
    font-size: 12px;
}

.InvalidInput {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 18px;
    font-weight: 500;
    color: #005C6E;
    margin: 0% 0% 1% 5%;
}

.ContainerStyle {
    margin-top: 10px;
    padding-top: 5%;
    padding-bottom: 5%;
    background-color: #31B7BC;
    position: center;
    left: 50%;
    width: 40% !important;
    top: 50%  !important;
}

.MobileContainerStyle {
    margin-top: 10px;
    padding-top: 5%;
    padding-bottom: 5%;
    background-color: #31B7BC;
    position: center;
    left: 50%;
    width: 90% !important;
    top: 250px !important;
}

.mobiletextbox input {
    width: 250px;
    text-align: center;
}

.mobiletextbox {
    width: 200px;
}

.mobiletextbox label {
    margin-left: 5px;
}
