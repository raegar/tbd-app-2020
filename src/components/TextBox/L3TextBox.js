import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

class L3TextBox extends React.Component {
	constructor(props) {
		super(props);

		this.props = props;
		this.state = {
			emptyInput: this.props.clearInput,
			input: '',
		};
	}

    handleChange = (e) => {
    	if (this.state.emptyInput) {
    		this.setEmptyInput(false);
    	}
        
    	this.setInput(e.currentTarget.value);
    	console.log(this.state.input);  
    }

    setEmptyInput = (input) => {
    	this.setState({emptyInput: input});
    }

    setInput = (input) => {
    	this.setState({input: input});
    }

    clearInputField = (e) => {
    	this.setEmptyInput(!this.props.clearInput);
    }

    render = () => {
    	const className = isMobile ? "mobiletextbox" : "textbox";
	
    	return (
    		<div className={`${className ? "mobiletextbox" : "textbox"}`} id={this.props.wrapID}>
    			<label htmlFor={this.props.id}>{this.props.title}</label>
    			<input
    				name={this.props.name}
    				placeholder={this.props.placeholder}
    				id={this.props.id}
    				key={this.props.key}
    				maxLength={this.props.maxLength}
    				minLength={this.props.minLength}
    				size={this.props.size}
    				type={this.props.type}
    				min={this.props.min}
    				max={this.props.max}
    				value={this.state.emptyInput ? '': this.state.input}
    				disabled={this.props.attribute}
    				onChange={this.handleChange}
    				clearInput={this.clearInputField}
    			/>
    		</div>
    	);
    }
}

L3TextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

L3TextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<L3TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default L3TextBox;
