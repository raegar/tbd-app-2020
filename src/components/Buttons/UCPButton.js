import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './UCPButton.css';

function UCPButton({to, className, buttonText, ...props}) {
	if (to === undefined) {
		to = "/emptyLink";
	} else if (to === "none") {
		return(
			<button data-testid="ucpButton"
				type="button"
				className={className}
				{...props}>
				{buttonText}​
			</button>
		);
	}  
	
	return (
		<Link to={to}>
			<button data-testid="ucpButton"
				type="button"
				className={className}
				{...props}>
				{buttonText}​
			</button>
		</Link>
	);
}

UCPButton.propTypes = {
	/**
   * Is this the principal call to action on the page?
   */
	className:PropTypes.string,
	/**
   * Optional click handler
   */
	onClick:PropTypes.func,
};

UCPButton.defaultProps = {
	size:'UCPDefault',
	onClick:undefined
};

export default UCPButton;

/* Example

<UCPButton to="NAMEOFPAGE" primary=true className="Large" buttonText="Click Here"/> 

*/


