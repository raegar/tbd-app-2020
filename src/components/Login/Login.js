import React from 'react';
import { Container, Row , Col} from 'react-bootstrap';
import TextBox from '../TextBox/LoginTextBox';
import UCPButton from '../Buttons/UCPButton';
import './Login.css';
import {isMobile} from 'react-device-detect';
import {  Redirect } from 'react-router';

const initialState = {
	username: "",
	password: "",
	usernameError: "",
	passwordError: "",
	location: ""
};

class LoginForm extends React.Component {
	constructor() {
		super();
		this.state = initialState;
	}

    handleChangeUsername = (event) => {
    	this.setState({username: event.target.value });
    }

    handleChangePassword = (event) => {
    	this.setState({password: event.target.value });
    }
    
    handleSubmit = (event) => {
    	event.preventDefault();
    	const isValid = this.validate();
    	if (isValid) {
    		this.setState(initialState);
    		this.setState({redirect: true});
    	}
    }

    validateEmail = (email) => {
    	const pattern = /[a-zA-Z0-9]+[.]?([a-zA-Z0-9]+)?[@][a-z]{3,9}[.][a-z]{2,5}/;
    	const result = pattern.test(email);

    	if (result) {
    		this.setState({
    			usernameError: false,
    			username: email,
    		});
    		return true;
    	} else {
    		this.setState({
    			usernameError: true,
    		});
    		return false;
    	}
    };

    validate = () => {
    	let usernameError = "";
    	let passwordError = "";

    	if (!this.validateEmail(this.state.username)) {
    		usernameError = "Invalid username!";
    	}

    	//TODO: make this more secure by only disclosing one or the other is wrong
    	if (!this.state.password) {
    		passwordError = "Invalid password!";   
    	}

    	if (usernameError || passwordError) {
    		this.setState({usernameError, passwordError});
    		return false;
    	}

    	if (this.state.password === "Password") {
    		global.staffFirstName = this.state.username.substring(0, this.state.username.indexOf('.'));
    		global.staffLastName = 
                this.state.username.substring(this.state.username.indexOf('.') + 1, this.state.username.indexOf('@'));
    		global.userType = "staff";
    		return true;
    	}

    	return false;
    };

    clearForm = () => {
    	this.setState(initialState);
    };

    render = () => {
    	if (this.state.redirect) {
    		return <Redirect push to="/NewApplicationPage" />;
    	}

    	const className = isMobile ? "MobileContainerStyle" : "ContainerStyle";

    	return(
    		<Container className={className}>
    			<form>
    				<Row>
    					<Col>
    						<div className="UserInfo">
    							<h2>Login</h2>
    						</div>
    					</Col>
    				</Row>
    				<Row>
    					<Col>
    						<div className="UserInfo">
    							<TextBox
    								type="text" 
    								title="Username" 
    								value={this.state.username} 
    								onChange={this.handleChangeUsername}
    								placeholder="Enter Your Username"
    							/> 
    						</div>
    						<div className="InvalidInput">
    							<span>{this.state.usernameError}</span>
    						</div>
    					</Col>
    				</Row>
    				<Row>
    					<Col>
    						<div className="UserInfo">
    							<TextBox 
    								title="Password" 
    								type="password" 
    								value={this.state.password} 
    								onChange={this.handleChangePassword} 
    								placeholder="Enter Your Password"
    							/> 
    						</div>
    						<div className="InvalidInput">
    							<span>{this.state.passwordError}</span>
    						</div>
    					</Col>
    				</Row>
    				<div className="UserInfo">
    					<UCPButton onClick={this.handleSubmit} to="none" className="smallbutton" buttonText="Submit"
    					/>                   
    				</div>
    			</form>
    		</Container>
    	);
    }
}

export default LoginForm;