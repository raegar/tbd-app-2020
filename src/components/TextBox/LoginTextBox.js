import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

function LoginTextBox(wrapID, id, title, name, onChange, placeholder, maxLength, minLength, size, type, min, max, attribute) {
	const className = isMobile ? "mobiletextbox" : "textbox";

	return (
		<div className={`${className ? "mobiletextbox" : "textbox"}`} id={wrapID}>
			<label htmlFor={id}>{title}</label>
			<input
				name={name}
				onChange={onChange}
				placeholder={placeholder}
				id={id}
				maxLength={maxLength}
				minLength={minLength}
				size={size}
				type={type}
				min={min}
				max={max}
				disabled={attribute}
			/>
		</div>
	);
}

LoginTextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

LoginTextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default LoginTextBox;
