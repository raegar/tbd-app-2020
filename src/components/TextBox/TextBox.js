import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

function TextBox({
	wrapID, 
	id, 
	title, 
	name, 
	onChange, 
	placeholder, 
	maxLength, 
	minLength, 
	size, 
	type, 
	min, 
	max, 
	attribute,
	multiline, 
	...props}) {
	const textBoxClassName = isMobile ? "mobiletextbox" : "textbox";
	/*TODO: can we cull a bunch of these via the use of {...props}? it's basically meant to
	be a copy of input so some poor soul could get points for adding a component anyways.*/
	const textBoxPropsObj = {
		name: name,
		onChange: onChange,
		placeholder: placeholder,
		id: id,
		maxLength: maxLength,
		minLength: minLength,
		size: size,
		type: type,
		min: min,
		max: max,
		disabled: attribute,
		...props
	};

	//<input> doesn't work with multiple lines 
	return (
		<div className={`${textBoxClassName ? "mobiletextbox" : "textbox"}`} id={wrapID}>
			<label htmlFor={id}>{title}</label>
			{(!multiline) ? 
				<input 
					{...textBoxPropsObj}
				/>
				: <textarea
					{...textBoxPropsObj}
				/>}
		</div>
	);
}

TextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

TextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default TextBox;
