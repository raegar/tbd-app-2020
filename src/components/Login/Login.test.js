import React from 'react';
import { render, cleanup, fireEvent, queryAllByPlaceholderText } from '@testing-library/react';
import '@testing-library/jest-dom';
import UCPButton from '../Buttons/UCPButton';
import Login from './Login';
import TextBox from '../TextBox/TextBox';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders Login", () => {
    const { asFragment } = render(<Login/>);
    expect(asFragment()).toMatchSnapshot();
});

it("Renders Submit Button", () => {
    const {queryByTestId, queryByPlaceholderName} = render(<UCPButton to="none" className="smallbutton" buttonText="Submit"/>)
    expect(queryByTestId("ucpButton")).toBeTruthy()

});

it("Renders Username TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
        type="text" title="Username" placeholder="Enter Your Username"/> )
    expect(queryByPlaceholderText("Enter Your Username")).toBeTruthy()   
});
it("Renders Password TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox title="Password" type="password"  placeholder="Enter Your Password"/>  )
    expect(queryByPlaceholderText("Enter Your Password")).toBeTruthy()   
});

describe("Input values", () => {
    it("updates username on change", () => {
        const {queryByPlaceholderText} = render(<TextBox placeholder="Enter Your Username"/>)

        const textInput = queryByPlaceholderText('Enter Your Username')

        fireEvent.change(textInput, {target : {value: "fName.lName@student.peterborough.ac.uk"}})
        
        expect(textInput.value).toBe("fName.lName@student.peterborough.ac.uk")
    });

    it("updates password on change", () => {
        const {queryByPlaceholderText} = render(<TextBox placeholder="Enter Your Password"/>)

        const textInput = queryByPlaceholderText('Enter Your Password')

        fireEvent.change(textInput, {target : {value: "Password"}})
        
        expect(textInput.value).toBe("Password")
    });
});

describe("User details", () => {
    describe("with no data inside query", () => {
        it("does not trigger handleSubmit function", () => {
            const handleSubmit = jest.fn();

            const {queryByTestId} = render(<Router><Login/></Router>)
            fireEvent.click(queryByTestId('ucpButton'))
            
            expect(handleSubmit).not.toHaveBeenCalled();
        })
    });

    describe("with data inside query", () => {
        it("triggers handleSubmit function", () => {
            const submitData = jest.fn();

            const {queryByTestId, queryByPlaceholderText} = render(<Router><Login/></Router>)

            const userInput = queryByPlaceholderText('Enter Your Username')
            const passInput = queryByPlaceholderText('Enter Your Password')

            fireEvent.change(userInput, {target : {value: "fName.lName@ucp.ac.uk"}})
            fireEvent.change(passInput, {target : {value: "Password"}})
            submitData[0] = userInput;
            submitData[1] = passInput;

            fireEvent.click(queryByTestId('ucpButton'))
            
            expect(submitData[0].value).toBe("fName.lName@ucp.ac.uk");
            expect(submitData[1].value).toBe("Password");
        })
    })
})