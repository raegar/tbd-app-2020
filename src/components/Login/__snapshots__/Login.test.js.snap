// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Login 1`] = `
<DocumentFragment>
  <div>
    <header>
      <a
        href="/tbd-app-2020"
      >
        <img
          alt="UCP official logo"
          class="UCPLogoBlue"
          data-testid="ucpLogoHeader"
          src="UCPLogo.png"
        />
      </a>
    </header>
    <div
      class="ContainerStyle container"
    >
      <form>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="UserInfo"
            >
              <h2>
                Login
              </h2>
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="UserInfo"
            >
              <div
                class="mobiletextbox"
              >
                <label>
                  Username
                </label>
                <input
                  maxlength="30"
                  minlength="0"
                  placeholder="Enter Your Username"
                  size="20"
                  type="text"
                />
              </div>
            </div>
            <div
              class="InvalidInput"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="UserInfo"
            >
              <div
                class="mobiletextbox"
              >
                <label>
                  Password
                </label>
                <input
                  maxlength="30"
                  minlength="0"
                  placeholder="Enter Your Password"
                  size="20"
                  type="password"
                />
              </div>
            </div>
            <div
              class="InvalidInput"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          class="UserInfo"
        >
          <button
            class="smallbutton"
            data-testid="ucpButton"
            type="button"
          >
            Submitâ€‹
          </button>
        </div>
      </form>
    </div>
    <footer
      class="ucpFooter"
    />
  </div>
</DocumentFragment>
`;
